tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,168 @@
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.extra_flags = None
+
+    s2nd = managed_process(S2N, server_options, timeout=5)
+    openssl = managed_process(provider, client_options, timeout=5)
+
+    for client_results in openssl.get_results():
+        client_results.assert_success()
+        assert_openssl_records_are_padded_correctly(
+            str(client_results.stdout), padding_size)
+
+    expected_version = get_expected_s2n_version(protocol, provider)
+
+    for server_results in s2nd.get_results():
+        server_results.assert_success()
+        # verify that the payload was correctly received by the server
+        assert random_bytes in server_results.stdout
+        # verify that the version was correctly negotiated
+        assert to_bytes("Actual protocol version: {}".format(
+            expected_version)) in server_results.stdout
+        # verify that the cipher was correctly negotiated
+        assert to_bytes("Cipher negotiated: {}".format(
+            cipher.name)) in server_results.stdout
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+# only tls 1.3 supports record padding
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("padding_size", PADDING_SIZES, ids=get_parameter_name)
+def test_s2n_client_handles_padded_records(managed_process, cipher, provider, curve, protocol, certificate,
+                                           padding_size):
+    port = next(available_ports)
+
+    client_random_bytes = data_bytes(PAYLOAD_SIZE)
+    server_random_bytes = data_bytes(PAYLOAD_SIZE)
+
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        cert=certificate.cert,
+        key=certificate.key,
+        insecure=True,
+        protocol=protocol,
+        data_to_send=server_random_bytes,
+        extra_flags=[
+            '-record_padding', padding_size]
+    )
+
+    client_options = copy.copy(server_options)
+    client_options.mode = Provider.ClientMode
+    client_options.extra_flags = None
+    client_options.data_to_send = client_random_bytes
+
+    # s2nc will wait until it has received the server's response before closing
+    s2nc = managed_process(S2N, client_options, timeout=5,
+                           close_marker=str(server_random_bytes)[2:-1])
+
+    # openssl will send its response after it has received s2nc's record
+    openssl = managed_process(provider, server_options,
+                              timeout=5, send_marker=str(client_random_bytes)[2:-1])
+
```

Comment:
Why are you ignoring the first two characters of your random data?

Reply:
The string output of the bytes have the form `b'<string>'`. Skipping the first two chars is for skipping the literal `b'`.

Reply:
I would suggest looking at how other tests that don't need to do [2:-1] handle the bytes / str conversions. I remember it's a bit of a mess, but doable. Maybe look at utils::to_string? If you can't figure it out, you need a comment and possible a little wrapper method.

Reply:
I played around with this a little.

You don't need to do this for the close_marker. You do need to do something like this for the send_marker when using openssl (but not s2n). We've gotten around this in different tests in different ways, [like my solution from the renegotiation test](https://github.com/aws/s2n-tls/blob/main/tests/integrationv2/test_renegotiate.py#L98-L102). But we should fix this at the framework level rather than repeatedly fighting with it in individual tests. Can you open an issue?

Reply:
Yes I'll open an issue